packages/perseus/src/interactive2/__tests__/movable.test.ts

```diff
@@ -0,0 +1,182 @@
+        expect(typeof movable.cursor).toBe("function");
+        expect(typeof movable.added).toBe("function");
+        expect(typeof movable.isHovering).toBe("function");
+        expect(typeof movable.isMouseOver).toBe("function");
+        expect(typeof movable.isDragging).toBe("function");
+        expect(typeof movable.mouseTarget).toBe("function");
+    });
+
+    it("cloneState()", () => {
+        const movable = new Movable(
+            new Graphie(document.createElement("div")),
+            {},
+        );
+
+        expect(movable.cloneState()).toStrictEqual(
+            expect.objectContaining({
+                id: expect.any(String),
+                add: [],
+                added: true,
+                cursor: null,
+                draw: [],
+                isDragging: false,
+                isHovering: false,
+                isMouseOver: false,
+                modify: [],
+                mouseTarget: null,
+                onClick: [],
+                onMove: [],
+                onMoveEnd: [],
+                onMoveStart: [],
+                remove: [],
+            }),
+        );
+    });
+
+    it("cloneState() should copy state", () => {
+        const movable = new Movable(
+            new Graphie(document.createElement("div")),
+            {},
+        );
+
+        expect(movable.cloneState()).not.toBe(movable.state);
+    });
+
+    it("modify() resets movable to initial state", () => {
+        const movable = new Movable(
+            new Graphie(document.createElement("div")),
+            {},
+        );
+
+        const state = movable.cloneState();
+        movable.state.added = false;
+
+        movable.modify();
+
+        expect(movable.cloneState()).toStrictEqual(state);
+    });
+
+    it("modify() retains original ID", () => {
+        const movable = new Movable(
+            new Graphie(document.createElement("div")),
+            {},
+        );
+        const id = movable.state.id;
+
+        movable.modify({added: true});
+
+        expect(movable.state.id).toStrictEqual(id);
+    });
+
+    it("grab() fires onMoveStart", () => {
+        const graphie = new Graphie(document.createElement("div"));
+        graphie.init({
+            range: [
+                [-10, 10],
+                [-10, 10],
+            ],
+        });
+        const onMoveStart = jest.fn();
+        const movable = new Movable(graphie, {onMoveStart});
+
+        movable.grab([0, 0]);
+
+        expect(onMoveStart).toHaveBeenCalledWith([0, 0], [0, 0]);
+    });
+
+    it("grab() fires onMove when mouse moves", () => {
+        const graphie = new Graphie(document.createElement("div"));
+        graphie.init({
+            range: [
+                [-10, 10],
+                [-10, 10],
+            ],
+        });
+        const onMove = jest.fn();
+        const movable = new Movable(graphie, {onMove});
+
+        movable.grab([0, 0]);
+        fireEvent.mouseMove(document, {clientX: 480, clientY: 320});
```

Comment:
I assume these coordinates are derived from the scale of the Graphie somehow? It might be nice to set the scale explicitly (to decouple this test from whatever defaults Graphie sets) or otherwise document where these numbers come from.

Reply:
Agreed. Updated.